### PR TITLE
Teamcity script refactoring

### PR DESCRIPTION
**Pull request by KmaTTi89**
Make Teamcity.ps1 more generic

### CODE REVIEW

**Comment by warpech** [comment]
@joozek78 could you pls review this? The purpose of this change it to make it possible to use the same script in all the projects. The apps to build and tests to build & run should be passed to it as input params.

File: tools/Teamcity.ps1 [hunk 1]
```diff
@@ -1,56 +1,91 @@
...
+{
+	$fileContent = "<?xml version=`"1.0`" encoding=`"UTF-8`"?><service><server-dir>$StarcounterRepoPath</server-dir></service>"
+	New-Item -Path $StarcounterConfigPath -Name personal.xml -ItemType "file" -force -Value $fileContent | Out-Null
+}
 
-$NunitConsoleRunnerExePath = "$checkoutdir\KitchenSink\packages\NUnit.ConsoleRunner.$nunitversion\tools\nunit3-console.exe"
-$NunitArg = "$KitchenSinkTestsPath --noheader --teamcity --params Browsers=$browsersToRun"
+Function createRepo()
+{
+	Start-Process -FilePath $StarExePath -ArgumentList "`@`@createrepo $StarcounterWorkDirPath" -NoNewWindow -Wait
```
Comment (joozek78) [inline tools/Teamcity.ps1]: why is only first `@` escaped?

File: tools/Teamcity.ps1 [hunk 2]
```diff
@@ -1,56 +1,91 @@
...
-$StarAdminExePath = "$StarCounterDir\staradmin.exe"
+Function runApps($apps, $source)
+{
+	foreach ($app in $apps)
+	{
+		$AppWWWPath = "$checkoutDir\$testedApp\$source\$app\wwwroot"
+		$AppExePath = "$checkoutDir\$testedApp\$source\$app\bin\Debug\$app.exe"
+		$AppArg = "--resourcedir=$AppWWWPath $AppExePath"
+		
+		$process = Start-Process -FilePath $StarExePath -ArgumentList $AppArg -PassThru -NoNewWindow
```
Comment (joozek78) [inline tools/Teamcity.ps1]: why not just `-Wait`?

File: tools/Teamcity.ps1 (hunk 1 above)
Comment (KmaTTi89) [inline tools/Teamcity.ps1]: I can see that second one is escaped too but maybe only on my monitor
The PowerShell escape character is the grave-accent(`)

File: tools/Teamcity.ps1 (hunk 2 above)
Comment (KmaTTi89) [inline tools/Teamcity.ps1]: I can provide id of the process to wait

File: tools/Teamcity.ps1 (hunk 1 above)
Comment (joozek78) [inline tools/Teamcity.ps1]: I get it now :) It looks like the first one is "quoted", but actually both are preceded with "`"

**Pull request merged**